essentials/src/main/java/solutions/Solution3.java

```diff
@@ -45,11 +49,20 @@ public static void main(String[] args) {
     private static Pipeline buildPipeline() {
         Pipeline p = Pipeline.create();
 
-        StreamStage<Trade> source = p.drawFrom(TradeSource.tradeSource())
-                .withNativeTimestamps(0);
+        p.drawFrom(TradeSource.tradeSource(1))
+         .withNativeTimestamps(0 )
+         .mapStateful(
```

Comment:
this should be keyed, otherwise you're not comparing trades for same key

Reply:
That's fine for this lab I guess. The intention is to detect a simple pattern.

Reply:
but how can it be considered a price drop if it's different stocks?